modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/security/DoubleSignDetectRunner.scala

```diff
@@ -0,0 +1,37 @@
+package org.tessellation.sdk.infrastructure.security
+
+import cats.effect.Async
+import cats.effect.std.Supervisor
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+
+import scala.concurrent.duration.DurationInt
+
+import org.tessellation.sdk.domain.Daemon
+import org.tessellation.sdk.domain.fork.ForkInfoStorage
+import org.tessellation.sdk.domain.security.{DoubleSignDetect, DoubleSignDetectRunner}
+
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+object DoubleSignDetectRunner {
+
+  val doubleSignDetectLoggerName = "DoubleSignDetectLogger"
+
+  def make[F[_]: Async](storage: ForkInfoStorage[F], detect: DoubleSignDetect[F]): DoubleSignDetectRunner[F] =
+    new DoubleSignDetectRunner[F] {
+
+      val logger = Slf4jLogger.getLoggerFromName[F](doubleSignDetectLoggerName)
+
+      override def run: F[Unit] = storage.getForkInfo.flatMap {
+        _.forks.toList.traverse_ {
+          case (id, entries) =>
+            detect.detect(id, entries).traverse_ { detection =>
```

Comment:
`detect.detect` should be imo `doubleSignDetector.detect` or better `doubleSignValidator.validate`

Reply:
Changed to

```
      override def run: F[Unit] = storage.getForkInfo.flatMap {
        _.forks.toList.traverse_ {
          case (id, entries) =>
            doubleSignValidator.validate(id, entries).traverse_ { detection =>
              logger.warn(s"Found a double signing: $detection")
            }
        }
      }
```